### PR TITLE
Update CloudFront documentation to implement use of preferred `aws_cloudfront_origin_access_control` for S3 access.

### PR DESCRIPTION
**Pull request by magnetikonline**
### Description

AWS have updated CloudFront with [Origin Access Controls (OAC)](https://aws.amazon.com/blogs/networking-and-content-delivery/amazon-cloudfront-introduces-origin-access-control-oac/) which now are considered a better option vs the previous origin access identity (OAI) method.

This PR updates the documentation/example for CloudFront to better reflect this use.

It also makes it clear that an `origin {}` block can contain one of `origin_access_control_id`, `s3_origin_config` (legacy OAI) or `custom_origin_config`.

### CODE REVIEW

File: website/docs/r/cloudfront_distribution.html.markdown [hunk 1]
```diff
@@ -45,11 +45,8 @@ locals {
 resource "aws_cloudfront_distribution" "s3_distribution" {
   origin {
     domain_name = aws_s3_bucket.b.bucket_regional_domain_name
+    origin_access_control_id = aws_cloudfront_origin_access_control.default.id
     origin_id   = locals.s3_origin_id
-
-    s3_origin_config {
-      origin_access_identity = "origin-access-identity/cloudfront/ABCDEFG1234567"
-    }
   }
```
Comment (magnetikonline) [inline website/docs/r/cloudfront_distribution.html.markdown:50]: Figured it was better to lead off the example to use the new `origin_access_control_id` method with an origin access control.

File: website/docs/r/cloudfront_distribution.html.markdown [hunk 2]
```diff
@@ -464,7 +461,7 @@ argument should not be specified.
 
 * `custom_origin_config` - The [CloudFront custom
     origin](#custom-origin-config-arguments) configuration information. If an S3
-    origin is required, use `s3_origin_config` instead.
+    origin is required, use `origin_access_control_id` or `s3_origin_config` instead.
```
Comment (magnetikonline) [inline website/docs/r/cloudfront_distribution.html.markdown:464]: This wasn't clear - if you specify a `origin_access_control_id` - you don't need to give a `s3_origin_config` block.

File: website/docs/r/cloudfront_distribution.html.markdown [hunk 3]
```diff
@@ -473,7 +470,7 @@ argument should not be specified.
     `value` parameters that specify header data that will be sent to the origin
     (multiples allowed).
 
-* `origin_access_control_id` (Optional) - The unique identifier of an origin access control for this origin.
+* `origin_access_control_id` (Optional) - The unique identifier of a [CloudFront origin access control][8] for this origin.
```
Comment (magnetikonline) [inline website/docs/r/cloudfront_distribution.html.markdown:473]: Felt a link to the `aws_cloudfront_origin_access_control` resource doc page was a good idea.

File: website/docs/r/cloudfront_distribution.html.markdown [hunk 4]
```diff
@@ -513,8 +510,7 @@ argument should not be specified.
 
 ##### S3 Origin Config Arguments
 
-* `origin_access_identity` (Optional) - The [CloudFront origin access
-  identity][5] to associate with the origin.
+* `origin_access_identity` (Required) - The [CloudFront origin access identity][5] to associate with the origin.
```
Comment (magnetikonline) [inline website/docs/r/cloudfront_distribution.html.markdown:513]: This is actually required - not optional.

**Review by bschaatsbergen — APPROVED** [review]
LGTM :rocket:

**Comment by magnetikonline** [comment]
> LGTM 🚀

Sorry @bschaatsbergen just gotta fix a `terraform fmt` issue - about to force push 👍 

thx for the approve 👍

**Comment by magnetikonline** [comment]
@bschaatsbergen just forced pushed - should fix the `terraform fmt` I messed up changing the CloudFront first usage example to include `origin_access_control_id`. 👍

**Review by bschaatsbergen — APPROVED** [review]
LGTM :rocket:

**Review by ewbankkit — APPROVED** [review]
LGTM :rocket:.

**Comment by ewbankkit** [comment]
@magnetikonline Thanks for the contribution :tada: :clap:.

**Pull request merged**